frontend/src/components/object/ObjectFilters.vue

```diff
@@ -0,0 +1,173 @@
+import { Permissions } from '@/utils/constants';
+
+import type { MetadataPair, Tag } from '@/types';
+
+// Props
+type Props = {
+  bucketId?: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  bucketId: undefined,
+});
+
+// Store
+const metadataStore = useMetadataStore();
+const objectStore = useObjectStore();
+const tagStore = useTagStore();
+const { getMetadataSearchResults } = storeToRefs(useMetadataStore());
+const { getUnfilteredObjectIds } = storeToRefs(useObjectStore());
+const { getTagSearchResults } = storeToRefs(useTagStore());
+
+// State
+const searching = ref(false);
+const selectedMetadata = ref([]);
+const selectedTags = ref([]);
+
+// Store subscriptions
+objectStore.$onAction(
+  ({name, args}) => {
+    // If someone calls fetchObjects to refresh the table, clear the filter
+    // unless supplied with filter(s)
+    if (name === 'fetchObjects') {
+      // Args are supplied from onAction as a single array so have to check how many to satisfy needed condition
+      // Might be a cleaner way of doing this...
+      if(args.length < 2) {
+        selectedMetadata.value = [];
+        selectedTags.value = [];
+      }
+    }
+  }
+);
+
+// Computed
+const metadataValues = computed(() => {
+  // Filter out any tags that don't have an objectID that exist in getObjects
+  const filteredVals = getMetadataSearchResults.value.filter((searchRes) =>
+    getUnfilteredObjectIds.value.some((obj) => obj === searchRes.objectId)
+  );
+
+  // Take the metadata for the objects, and flatten them into a single array
+  const metadataVals = [
+    ...new Set(filteredVals.flatMap((obj) => obj.metadata)),
+  ];
+  return metadataVals
+    // Add a display property to each tag to be used by the multiselect
+    .map((val) => ({ ...val, display: `${val.key}:${val.value}` }))
+    // Unique by display property
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+const tagsetValues = computed(() => {
+  // Filter out any tags that don't have an objectID that exist in getUnfilteredObjectIds
+  const filteredVals = getTagSearchResults.value.filter((searchRes) =>
+    getUnfilteredObjectIds.value.some((obj) => obj === searchRes.objectId)
+  );
+
+  // Take the tags for the objects, and flatten them into a single array
+  const taggingVals = [
+    ...new Set(filteredVals.flatMap((obj) => obj.tagset)),
+  ];
+  return taggingVals
+    // Add a display property to each tag to be used by the multiselect
+    .map((val) => ({ ...val, display: `${val.key}=${val.value}` }))
+    // coms-id not allowed as a tag to filter on by COMS
+    .filter((val) => val.key !== 'coms-id')
+    // Unique by display property
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
```

Comment:
This is very duplicate but my attempts to DRY it kept getting difficulties with TypeScript. If anyone has any good ideas I'm all ears